packages/origin-ui/src/components/DemandForm.tsx

```diff
@@ -736,9 +900,12 @@ class OnboardDemandClass extends React.Component<IStateProps, IState> {
     }
 }
 
-export const OnboardDemand = connect(
-    (state: IStoreState): IStateProps => ({
-        currentUser: getCurrentUser(state),
-        configuration: getConfiguration(state)
-    })
-)(OnboardDemandClass);
+export const DemandForm = withRouter(
+    connect(
+        (state: IStoreState): IStateProps => ({
+            baseURL: getBaseURL(state),
+            currentUser: getCurrentUser(state),
+            configuration: getConfiguration(state)
+        })
+    )(DemandFormClass)
+);
```

Comment:
This file has become huge, is there any way to separate it to make it more readable?

Reply:
Refactored